onnxruntime/core/session/default_cpu_allocator_c_api.cc

```diff
@@ -23,10 +23,33 @@ struct OrtDefaultAllocator : OrtAllocatorImpl {
   ~OrtDefaultAllocator() override { OrtReleaseAllocatorInfo(cpuAllocatorInfo); }
 
   void* Alloc(size_t size) {
-    return ::malloc(size);
+    if (size == 0)
+      return nullptr;
+    //default align to 64;
+    void* p;
+#if defined(_WIN32) && !defined(_WIN64)
+  size_t alignment = 32;
+#else  
+  size_t alignment = 64;
+#endif
+#if _MSC_VER
+    p = _aligned_malloc(size, alignment);
+    if (p == nullptr) throw std::bad_alloc();
```

Comment:
Are there any scenarios where aligned allocation is not the best choice? e.g. is it a good option on a device with limited memory?

Reply:
As long as tensor size >> alignment, there should be no issue with this. Besides, this default allocator could be replaced by user defined allocator if there's really that case. IMO the more common scenario is to return aligned address to providers for better performance.

---
In reply to: [312421603](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312421603) [](ancestors = 312421603)